src/sys/resource.rs

```diff
@@ -0,0 +1,230 @@
+                /// BSD specific Resource https://www.freebsd.org/cgi/man.cgi?query=setrlimit
+                #[cfg(not(any(target_os = "netbsd", target_os = "freebsd")))]
+                RLIMIT_AS,
+                RLIMIT_CORE,
+                RLIMIT_CPU,
+                RLIMIT_DATA,
+                RLIMIT_FSIZE,
+                RLIMIT_NOFILE,
+                RLIMIT_STACK,
+
+                // platform specific
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_KQUEUES,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_LOCKS,
+
+                #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "openbsd", target_os = "linux"))]
+                RLIMIT_MEMLOCK,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_MSGQUEUE,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_NICE,
+
+                #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "openbsd", target_os = "linux"))]
+                RLIMIT_NPROC,
+
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_NPTS,
+
+                #[cfg(any(target_os = "android", target_os = "freebsd", target_os = "openbsd", target_os = "linux"))]
+                RLIMIT_RSS,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_RTPRIO,
+
+                #[cfg(any(target_os = "linux"))]
+                RLIMIT_RTTIME,
+
+                #[cfg(any(target_os = "android", target_os = "linux"))]
+                RLIMIT_SIGPENDING,
+
+                #[cfg(any(target_os = "freebsd", target_os = "dragonfly"))]
+                RLIMIT_SBSIZE,
+
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_SWAP,
+
+                #[cfg(target_os = "freebsd")]
+                RLIMIT_VMEM,
+            }
+        }
+    }else{
+        // unkown os
+    }
+}
+
+/// Get the current processes resource limits
+///
+/// A value of `None` indicates that there's no limit.
+///
+/// # Parameters
+///
+/// * `resource`: The [`Resource`] that we want to get the limits of.
+///
+/// # Examples
+///
+/// ```
+/// # use nix::sys::resource::{getrlimit, Resource};
+///
+/// let (soft_limit, hard_limit) = getrlimit(Resource::RLIMIT_NOFILE).unwrap();
+/// println!("current soft_limit: {:?}", soft_limit);
+/// println!("current hard_limit: {:?}", hard_limit);
+/// ```
+///
+/// # References
+///
+/// [getrlimit(2)](https://linux.die.net/man/2/getrlimit)
+///
+/// [`Resource`]: enum.Resource.html
+pub fn getrlimit(resource: Resource) -> Result<(Option<rlim_t>, Option<rlim_t>)> {
+    let mut old_rlim = rlimit {
+        rlim_cur: 0,
+        rlim_max: 0,
+    };
+
+    cfg_if! {
+        if #[cfg(all(target_os = "linux", target_env = "gnu"))]{
+            let res = unsafe { libc::getrlimit(resource as __rlimit_resource_t, &mut old_rlim) };
```

Comment:
Thank you for pointing this out @asomers . I think this issue originated in [here](https://gcc.gnu.org/legacy-ml/gcc/2015-08/msg00441.html) (at least the resource I can find). So for linux/gnu `libc` bindings, it will use `u32` instead of `i32`. This is implemented in [libc](https://github.com/rust-lang/libc/blob/master/src/unix/linux_like/linux/gnu/mod.rs) as well.  Other linux env like musl will still remains as `i32` and that's why I used `cfg_if!`

Reply:
Ok, so it's a Linux man page bug.